src/components/ec-datepicker/ec-datepicker.vue

```diff
@@ -147,19 +166,40 @@ export default {
     }
   },
   methods: {
+    getListeners() {
+      const listeners = { ...this.$listeners };
+      delete listeners['value-change'];
+      delete listeners.open;
```

Comment:
Out of curiosity, why did we need to remove all these listeners?

Reply:
$listeners property can forward all event listeners on the component to a specific child element and they were causing conflicts in our ec-datepicker because we already have these events coming from the flatpickr instance, so the events we have defined in the options (mergeWithDefaultOptions method) which are ready, change, open and close we removed it. If for example, we do blur in our datepicker that event will be forwarded to our ec-input-field, we did not remove it, but all the events we have defined in our datepicker options should be removed from $listeners, the onChange was causing issues because first, we do this.$emit('value-change', d); passing directly a Date instance but we also have $listeners and the value-change will be executed again with the value of the input which is formattedValue (string type) setting the v-model to a string instead of having a Date instance, so we don't want to listen the onChange event coming from the ec-input-field component.